### PR TITLE
Merge Symfony 4 branch to master

### PR DESCRIPTION
**Pull request by ostrolucky**
Bc break, tag as 2.0. Optionally, I can create BC layer

### CODE REVIEW

**Review by klinec — APPROVED** [review]

File: src/Everlution/EnumBundle/DBAL/EnumDBType.php [hunk 1]
```diff
@@ -30,16 +30,16 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
     {
         $platform->markDoctrineTypeCommented($this);
 
-        return $platform->getSmallIntTypeDeclarationSQL($fieldDeclaration);
+        return $platform->getVarcharTypeDeclarationSQL($fieldDeclaration);
```
Comment (ivanbarlog) [inline src/Everlution/EnumBundle/DBAL/EnumDBType.php:33]: why are we changing SamllInt to Varchar? The whole point was to use integer instead of text...

Comment (ostrolucky) [reply src/Everlution/EnumBundle/DBAL/EnumDBType.php:33]: Whole point of storing it as integer was wrong as I mentioned originally. Numbers have nothing to do with what they represent. But nothing is stopping you to keep using them.

Comment (ivanbarlog) [reply src/Everlution/EnumBundle/DBAL/EnumDBType.php:33]: So what will be stored in database? string or integer?

Comment (ostrolucky) [reply src/Everlution/EnumBundle/DBAL/EnumDBType.php:33]: string

**Review by mdemo — CHANGES_REQUESTED** [review]
This is definitelly a step back. The support for 4.0 is ok...changing "int" to "string" can have impact on performance. Please revert, keeping only support for 4.0

**Comment by ostrolucky** [comment]
No performance impact https://stackoverflow.com/a/39650204

Integers are totally inapropriate for this:
https://stackoverflow.com/a/23554731
https://stackoverflow.com/a/20563821

We will keep using this branch then. Feel free to cherry-pick just Symfony 4 support commit.

File: src/Everlution/EnumBundle/DBAL/EnumDBType.php (hunk 1 above)
Comment (ivanbarlog) [reply src/Everlution/EnumBundle/DBAL/EnumDBType.php:33]: Ok, I am definitely not going to merge functionality which downgrades the performance.

**Pull request closed without merging**